sql-plugin/src/main/spark311/scala/org/apache/spark/sql/rapids/shims/RapidsHadoopWriterUtils.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*** spark-rapids-shim-json-lines
+{"spark": "311"}
+{"spark": "312"}
+{"spark": "313"}
+{"spark": "320"}
+{"spark": "321"}
+{"spark": "321cdh"}
+{"spark": "321db"}
+{"spark": "322"}
+{"spark": "323"}
+{"spark": "330"}
+{"spark": "330cdh"}
+{"spark": "330db"}
+{"spark": "331"}
+spark-rapids-shim-json-lines ***/
+package org.apache.spark.sql.rapids.shims
+
+import org.apache.hadoop.mapred.JobID
+
+object RapidsHadoopWriterUtils {
+  // SPARK-41448 create a jobID directly from the jobTrackerID
```

Comment:
The before and after SPARK-41448 have an additional defensive [check](https://github.com/apache/spark/blob/47ddac13144b5cf62464c3a4c02d5110ea72862a/core/src/main/scala/org/apache/spark/internal/io/SparkHadoopWriterUtils.scala#L62-L64) for negative `id`. Probably robuster to just follow their logic?

Reply:
Let me file a follow up PR to add this check.

Reply:
See https://github.com/NVIDIA/spark-rapids/pull/8458